server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -2730,6 +2730,44 @@ public void updateTabletDataFile(long maxCommittedTime, FileRef newDatafile, Fil
 
   }
 
+  /**
+   * Handles updating a tablets last Location
+   */
+  public void updateLastLocation(){
+    //unsure if this sync block is needed here. Need input.
+    synchronized (timeLock){
+    persistedTime = tabletTime.getAndUpdateTime();
+
+      MasterMetadataUtil.updateLastLocation(getTabletServer().getContext(), extent, tabletTime.getMetadataTime(persistedTime),
+                tabletServer.getClientAddressString(), tabletServer.getLock(),
+                lastLocation);
+
+      lastLocation = null;
+    }
+  }
+
+  /**
+   * Checks if a tablets last location needs to be updated
+   */
+  public boolean needsLastUpdate() {
+    boolean needsUpdate = false;
+    if (isClosing() || isClosed()) {
+      return false;
+    }
+    
+    Tablet t = tabletServer.getOnlineTablet(extent);
+    //will keep looking for a better way to check if a tablet is old and needs to be updated or not. In the mean time, this works and is adjustable.
+    long currentTime = tabletTime.getAndUpdateTime();
```

Comment:
tabletTime is not suitable for this type of timing information.  It tracks the time stamp that should be placed in keys and its implementation could just be a one 64bit counter that has nothing to do with wall clock time.

Reply:
Good to know, I will look for an alternative solution.

Reply:
I made some changes to use System time for currentTime. I also made a few other refactors. Do you think keeping oldTime the way it is okay or do I need to have a better way of tracking a tablet's age?